### ISSUE TITLE
Rendering tiff file problem

### ISSUE THREAD

**Issue by Destanek** [mapnik/mapnik#3412]
I have rendered with mapnik style earlier and have no problem with it
![0](https://cloud.githubusercontent.com/assets/17759444/14815001/17745e5e-0bd3-11e6-8d85-95e617134856.png)

but when I use MapQuest Style which include mq_world_mercator_x.tif at the low level zoom the output is this
![0](https://cloud.githubusercontent.com/assets/17759444/14815048/51a07784-0bd3-11e6-9834-1f4a374e974c.png)

the source tif file look like this
![world](https://cloud.githubusercontent.com/assets/17759444/14815242/72eebde6-0bd4-11e6-8e2d-408eb2f8bf0c.png)

you can see a xml file of layer-base.xml in [here](https://github.com/MapQuest/MapQuest-Mapnik-Style/blob/master/mapquest_inc/layer-base.xml.inc)
This problem only cause when render a low level zoom that include these tif file 

by the way I use generate_tile.py to render and change the render_size to 512 I don't know If it's cause this problem or not but it's work with mapnik style

**Comment by PhilHa** [comment]
I've got the same problem.
Have you found a solution already?

**Comment by Destanek** [comment]
No, I haven't found yet. I tried to change the Projection but it didn't work so I don't have any idea for now.

**Comment by lightmare** [comment]
Looks like an issue with flipped tile y-coordinates. Where can I get the failing tif? The link at https://github.com/MapQuest/MapQuest-Mapnik-Style doesn't work.

**Comment by Destanek** [comment]
You can download it from here 
[http://aweble.de/downloads/mercator_tiffs.tar.bz2](url)

**Comment by PhilHa** [comment]
But it's not just a reversed y-order.
On Zoomlevel 4 and 5 there are slipped stripes (and correct y-order apart from that).

Zoom 4:
![zoom4](https://cloud.githubusercontent.com/assets/5525659/17920074/6adae330-69d2-11e6-8e61-2fe83d27c86a.PNG)

Zoom 5:
![zoom5](https://cloud.githubusercontent.com/assets/5525659/17920076/6eef6b44-69d2-11e6-92f4-b82580c7a540.PNG)

**Comment by lightmare** [comment]
Small update. This issue is caused by wrong row indices calculated in `tiff_reader::read_stripped`. It's easy to fix, but I was silly enough to not "just squash the bug". I *had* to dig deeper :microscope: So I've been studying the intricacies of libtiff and constructing tricky images. More about that when I have a PR ready.

We actually have similarly broken reference images in test-data-visual. But our source images in test-data "play with this bug": some have only one strip, others' height is a multiple of rows/strip, and some are so subtle that the bug is not clearly visible. Steps to reproduce with an existing visual test, no external data:
```sh
gdal_translate -co BLOCKYSIZE=122 test/data/raster/river_merc.tiff /tmp/river_merc.tiff
mv /tmp/river_merc.tiff test/data/raster/
test/visual/run --agg --styles=tiff-opaque-edge-raster2
```
